website_sale_hide_price/models/product_template.py

```diff
@@ -38,3 +38,18 @@ def _get_combination_info(
             }
         )
         return combination_info
+
+    def _search_render_results(self, fetch_fields, mapping, icon, limit):
+        res = super()._search_render_results(fetch_fields, mapping, icon, limit)
+        new_result = []
+        for data in res:
+            product = self.env["product.template"].browse(data.get("id"))
+            if product.website_hide_price:
+                data.update(
+                    {
+                        "price": "<span>%s</span>"
+                        % (product.website_hide_price_message or "")
```

Comment:
Should we be aware of list_price as well?

Reply:
It seems to me that it could cause problems if that value is used for some later computation, it is better to just hide the legend for the end user

Reply:
Done! You are right should be erased in same way